### PR TITLE
feat(account-lib): add key factory for coins

### PR DESCRIPTION
**Pull request by byhow**
Ticket: STLX-10821

### CODE REVIEW

File: modules/account-lib/src/keyPair/index.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+export function register(coinName: string, source?: KeyPairOptions): BaseKeyPair {
+  const sanitizedCoinName = coinName.trim().toLowerCase();
+  const key = Object.keys(coinModules)
+    .filter((k) => coinModules[k].KeyPair)
+    // TODO(BG-40990): eth2 BLS keypair init error
+    .find((k) => k.trim().toLowerCase() !== 'eth2' && k.trim().toLowerCase() === sanitizedCoinName);
+  if (key) {
+    return new coinModules[key].KeyPair(source);
+  }
+  return {} as BaseKeyPair;
```
Comment (jrafael-simtlix) [inline modules/account-lib/src/keyPair/index.ts:14]: If a coin is not find, do we return empty object?
How does user handle this case?
Also there are no test case about this behavior.

**Review by 0xabrarhussain — CHANGES_REQUESTED** [review]
this change seems pretty wonky. this makes the convention assumption that everything in the folder above only has coin modules with the same structure

which sure you can have that convention, but it wont' be enforced

**Comment by byhow** [comment]
@0xabrarhussain The structure is not enforced in that we will implement the interface gradually if such structure is needed. For an initial factory method that gets the corresponding `key` implementation of each coin, we are not looking to refactor all of the previous coins' export structure.

For the exports module, it is checking if the structure exported the baseKey class implementation, so it does not need to follow the same structure, but only need to export that specific `key` module.

That being said, we are open to better ways to group the `key` class implementations without refactoring the old codebase for sure.

**Comment by byhow** [comment]
ref on the additional unit tests audits: https://github.com/BitGo/BitGoJS/pull/1832

**Comment by byhow** [comment]
@0xabrarhussain Do you mind we move this conversation over to #1832? And then we can close this in favor of the other PR

**Comment by byhow** [comment]
closing in favor of the other PR

**Pull request closed without merging**